src/main/groups.ts

```diff
@@ -0,0 +1,201 @@
+import { loadMapFromScriptProperties, setScriptProperty } from "./propertiesService";
+import { getCellValues } from "./scan";
+
+const COMMA_DELIMITER: string = ",";
+const GROUPS_TAB_NAME: string = "Groups";
+const SUPERGROUPS_TAB_NAME: string = "Supergroups";
+const GROUP_NAME_COLUMN_INDEX: number = 0;
+const GROUP_MEMBER_NAMES_COLUMN_INDEX: number = 1;
+const SUPERGROUP_NAME_COLUMN_INDEX: number = 0;
+const SUBGROUP_COLUMN_INDEX: number = 1;
+const GROUPS_MAP_KEY: string = "GROUPS_MAP";
+
+// Interface declared here because this is an internal object used only during the Supergroup loading process
+declare interface Supergroup {
+    name: string,
+    subgroups: string[],
+};
+
+// Maps a Supergroup name to its corresponding Supergroup object
+// Only used internally within this module for loading
+type SupergroupMap = { [key: string]: Supergroup };
+
+export const GROUPS_MAP: GroupsMap = loadMapFromScriptProperties(GROUPS_MAP_KEY) as GroupsMap;
+
+/**
+ * Loads groups and supergroups from the Onestop into script properties
+ */
+export function loadGroupsFromOnestopIntoScriptProperties(): void {
+    const groupsMap: GroupsMap = loadGroupsFromOnestop();
+    const supergroupsMap: GroupsMap = loadSupergroupsFromOnestop(groupsMap);
+    const combinedGroupsMaps: GroupsMap = mergeGroupsMaps(groupsMap, supergroupsMap);
+    const duplicateFreeGroupsMap: GroupsMap = removeDuplicatesFromGroupMaps(combinedGroupsMaps);
+
+    setScriptProperty(GROUPS_MAP_KEY, JSON.stringify(duplicateFreeGroupsMap));
+}
+
+function loadGroupsFromOnestop(): GroupsMap {
+    const cellValues: any[][] = getCellValues(GROUPS_TAB_NAME);
+    const groupsMap: GroupsMap = {};
+
+    // Start at row 1 to skip the table header row
+    for(let i = 1; i < cellValues.length; i++) {
+        const rowValues: any[] = cellValues[i];
+        const groupName: string = rowValues[GROUP_NAME_COLUMN_INDEX];
+        const groupMemberNames: string[] = getGroupMemberNames(rowValues);
+
+        if(groupName === "") {
+            // Skip empty entries
+            continue;
+        }
+
+        if(groupsMap.hasOwnProperty(groupName)) {
+            Logger.log(`Group ${groupName} has already been defined in the Groups table. Combining the two lists of members`);
```

Comment:
maybe we can prefix these kind of logs with something like "WARN:", since we don't have log levels available for us

Reply:
Added the `WARN` prefix